### ISSUE TITLE
how to exclude folders ?

### ISSUE THREAD

**Issue by rubiin** [gilamran/tsc-watch#67]
I am using tsc watch with nrest js application and i have a module which uploads a file. However when i upload a file, tsc watch restarts the nestjs application

**Comment by gilamran** [comment]
I'm guessing that your `tsconfig.json` includes all files... and I'm also guessing that you are uploading to a location where your `tsconfig.json` is looking at.
If I was correct with my guesses, you should exclude the upload folder in your `tsconfig.json`

**Comment by l2ubin** [comment]
i tried putting the upload folder which is in the root directory to the excluded list, however the problem persists. It works on nodemon though

**Comment by gilamran** [comment]
Can you please provide a simple example?
No need for the upload thing to work, I'll just create a new file manually.

**Comment by l2ubin** [comment]
You can check the  example on https://docs.nestjs.com/techniques/file-upload

**Comment by gilamran** [comment]
Sorry, I'm unfamiliar with nestjs, and the examples are not helping me reproduce your issue.
Can you try to use `typescript --watch` and see if the upload causes a recompilation?

**Comment by zhbhun** [comment]
Same to me. If one folder make new subfolder, tsc-watch would detect the change even though exclude that folder.

**Comment by rubiin** [comment]
decided to switch to nodemon as this does not occur on there

**Comment by gilamran** [comment]
I'm sorry, the watch functionality is done by Typescript itself.

**Comment by kotuck** [comment]
Hey guys. You just need to add `include` to `tsconfig`.

```
"include": ["src"],
"exclude": ["node_modules", "dist"]
```

**Comment by sduduzog** [comment]
@gilamran is it possible to specify a tsconfig like `--project tsconfig.api.json`

**Comment by gilamran** [comment]
TL; DR Yes

`tsc-watch` is running `typescript` with ALL your parameters (Only adds `--watch`), so everything that you can do with `typescript`, is possible with `tsc-watch`.

**Comment by sanzhardanybayev** [comment]
> Hey guys. You just need to add `include` to `tsconfig`.
> 
> ```
> "include": ["src"],
> "exclude": ["node_modules", "dist"]
> ```



> Hey guys. You just need to add `include` to `tsconfig`.
> 
> ```
> "include": ["src"],
> "exclude": ["node_modules", "dist"]
> ```

Thank you bro! You saved my day!
